From c50b42cb67c5b7fafa4c9ec6edfe577feccd4eec Mon Sep 17 00:00:00 2001
From: Gabriel Ravier <gabravier@gmail.com>
Date: Mon, 12 Sep 2022 16:52:08 +0200
Subject: [PATCH] Added %t conversion specifier to printf
To: dietlibc@fefe.de

The C standard states:

t Specifies that a following b, d, i, o, u, x, or X conversion specifier applies to a ptrdiff_t
or the corresponding unsigned integer type argument; or that a following n conversion
specifier applies to a pointer to a ptrdiff_t argument.

This patch accordingly implements this (with the hopefully accurate
assumption that sizeof(ptrdiff_t) == sizeof(size_t)) and makes it so software
relying on this flag works properly
---
 lib/__v_printf.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/__v_printf.c b/lib/__v_printf.c
index 536e837e..f2456855 100644
--- a/lib/__v_printf.c
+++ b/lib/__v_printf.c
@@ -102,6 +102,7 @@ inn_printf:
       case 'j':
 #endif
       case 'z':
+      case 't':
       case 'l':
 	++flag_long;
 	goto inn_printf;
-- 
2.37.3

